pkg/prometheus/statefulset.go

```diff
@@ -518,3 +519,100 @@ func GetStatupProbePeriodSecondsAndFailureThreshold(cfp monitoringv1.CommonProme
 
 	return int32(startupPeriodSeconds), int32(startupFailureThreshold)
 }
+
+func MakeContainerPorts(cpf monitoringv1.CommonPrometheusFields) []v1.ContainerPort {
+	return []v1.ContainerPort{
+		{
+			Name:          cpf.PortName,
+			ContainerPort: 9090,
+			Protocol:      v1.ProtocolTCP,
+		},
+	}
+}
+
+func BuildVolumes(
+	p monitoringv1.PrometheusInterface,
+	tlsSecrets *operator.ShardedSecret,
+) ([]v1.Volume, []v1.VolumeMount, []v1.VolumeMount, error) {
+	volumes, promVolumeMounts, err := BuildCommonVolumes(p, tlsSecrets)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	configReloaderVolumeMounts := CreateConfigReloaderVolumeMounts()
+
+	return volumes, promVolumeMounts, configReloaderVolumeMounts, nil
+}
+
+func CreateConfigReloaderVolumeMounts() []v1.VolumeMount {
+	return []v1.VolumeMount{
+		{
+			Name:      "config",
+			MountPath: ConfDir,
+		},
+		{
+			Name:      "config-out",
+			MountPath: ConfOutDir,
+		},
+	}
+}
+
+func BuildWebconfig(
+	cpf monitoringv1.CommonPrometheusFields,
+	p monitoringv1.PrometheusInterface,
+) (monitoringv1.Argument, []v1.Volume, []v1.VolumeMount, error) {
+	var fields monitoringv1.WebConfigFileFields
+	if cpf.Web != nil {
+		fields = cpf.Web.WebConfigFileFields
+	}
+
+	webConfig, err := webconfig.New(WebConfigDir, WebConfigSecretName(p), fields)
+	if err != nil {
+		return monitoringv1.Argument{}, nil, nil, err
+	}
+
+	confArg, configVol, configMount, err := webConfig.GetMountParameters()
+	if err != nil {
+		return monitoringv1.Argument{}, nil, nil, err
+	}
+
+	return confArg, configVol, configMount, nil
+}
+
+// The /-/ready handler returns OK only after the TSDB initialization has
+// completed. The WAL replay can take a significant time for large setups
+// hence we enable the startup probe with a generous failure threshold (15
+// minutes) to ensure that the readiness probe only comes into effect once
+// Prometheus is effectively ready.
+// We don't want to use the /-/healthy handler here because it returns OK as
+// soon as the web server is started (irrespective of the WAL replay).
+func MakeProbes(
```

Comment:
As @simonpasquier said in https://github.com/prometheus-operator/prometheus-operator/pull/6652#discussion_r1637887343 we could have either same settings with statefulset or tighter timeouts for daemonset. Let's have a decision here, because it affects the refactoring?

Reply:
This probably would affect how we'll refactor `TestStartupProbeTimeoutSeconds` too? https://github.com/prometheus-operator/prometheus-operator/blob/cbf0f66c687656b6a96e2acb8b618ef0aaaf138c/pkg/prometheus/agent/statefulset_test.go#L193

Reply:
Until we're sure which direction to go (different probe settings or not), let's delay the refactoring then.

Reply:
This piece is already a duplication in server and agent's statefulsets, so I think just keep this refactoring here for now? Later if we decide to change the timeouts for daemonset we can modify this refactoring to work with all three modes.

Reply:
I refactored `TestStartupProbeTimeoutSeconds` in #6694 too. We still need this test for DaemonSet now, so I think we should refactor to reduce duplication anyway. We can modify the refactored function later if we decide to have different timeouts for DaemonSet.